[AArch64] Push mul into extend operands

davemgreen opened the pull request:
In a similar way to how we push vector adds into extends, this pushed 'mul(zext,zext)' into 'zext(mul(zext,zext))' if the extend can be done in two or more steps.

https://alive2.llvm.org/ce/z/WjU7Kr

SamTebbs33 reviewed (approved):
Looks simple and effective to me. Thanks.

davemgreen committed:
[AArch64] Push mul into extend operands

In a similar way to how we push vector adds into extends, this pushed
'mul(zext,zext)' into 'zext(mul(zext,zext))' if the extend can be done
in two or more steps.

https://alive2.llvm.org/ce/z/WjU7Kr
